lib/pages/edit_page/bloc/edit_page_state.dart

```diff
@@ -2,6 +2,13 @@ part of 'edit_page_bloc.dart';
 
 @immutable
 class EditPageState extends Equatable {
+  EditPageState.connection(Connection connection)
+      : name = connection.name,
+        accessKey = connection.accessKey,
+        secretKey = connection.secretKey,
+        endpoint = connection.endpoint,
+        bucket = connection.bucket;
+
```

Comment:
Why another constructor?

Reply:
In order not to break the old semantics and explicitly determine that the state is created based on an existing connection.